engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/IdentityLinkEntity.java

```diff
@@ -257,4 +257,26 @@ public String toString() {
            + ", tenantId=" + tenantId
            + "]";
   }
+
+  /* (non-Javadoc)
+   * @see java.lang.Object#clone()
+   */
+  public IdentityLinkEntity clone() {
+    IdentityLinkEntity copy = new IdentityLinkEntity();
+
+    copy.setId(this.getId());
+    copy.setType(this.getType());
+    copy.setUserId(this.getUserId());
+    copy.setGroupId(this.getGroupId());
+    copy.setTaskId(this.taskId);
+    copy.setProcessDefId(this.getProcessDefId());
+    copy.setTenantId(this.getTenantId());
+    copy.setTask(this.getTask());
+    ProcessDefinitionEntity processDef = this.getProcessDef();
+    if (processDef != null) {
+      copy.setProcessDef(processDef);
+    }
+
+    return copy;
```

Comment:
Is there an alternative solution to avoid copying the attributes? If we add new attributes in the future, we might easily miss adding them to the `#clone` method.

Reply:
I checked it again and it looks like we are not altering the existing identity links. Maybe it's enough to create a new list from `task.getIdentityLinks()`'s result instead of cloning/copying the entities?

Reply:
So far my testing showed that we need a copy. I will double check.

Reply:
@yanavasileva, we could do a sparring session on other solution ideas if you like.